geomstats/visualization.py

```diff
@@ -490,6 +494,186 @@ def draw(self, ax, **kwargs):
+                             rstride=1, cstride=1, color='grey',
+                             linewidth=0, alpha=.1, zorder=-1)
+        self.ax.plot_wireframe(coords_x, coords_y, coords_z, linewidths=.6,
+                               color='grey', alpha=.6, zorder=-1)
+
+        def lim(theta):
+            return gs.pi - self.elev + (2 * self.elev - gs.pi) / gs.pi * abs(
+                self.azim - theta)
+
+        for theta in gs.linspace(0, 2 * gs.pi, n_theta // 2 + 1):
+            for phi in gs.linspace(0, gs.pi, n_phi):
+                if theta <= self.azim + gs.pi and phi <= lim(theta):
+                    self.draw_triangle(theta, phi, scale)
+                if theta > self.azim + gs.pi and phi \
+                        < lim(2 * self.azim + 2 * gs.pi - theta):
+                    self.draw_triangle(theta, phi, scale)
+
+    def draw_triangle(self, theta, phi, scale):
+        """Draw the corresponding triangle on the sphere at theta, phi."""
+        u_theta = gs.cos(theta) * self.ua + gs.sin(theta) * self.na
+        T = gs.cos(phi / 2) * self.can + gs.sin(phi / 2) * u_theta
+        T = scale * T
+        T = gs.hstack((T, .5 * gs.ones((3, 1))))
+        T = self.R_theta_phi(theta, phi) @ T.transpose()
+
+        x = gs.hstack((T[0], T[0, 0]))
+        y = gs.hstack((T[1], T[1, 0]))
+        z = gs.hstack((T[2], T[2, 0]))
+
+        self.ax.plot3D(x, y, z, 'grey', zorder=1)
+        c = ['red', 'green', 'blue']
+        for i in range(3):
+            self.ax.scatter(x[i], y[i], z[i],
+                            color=c[i], s=10, alpha=1, zorder=1)
+
+    @staticmethod
+    def R_theta_phi(theta, phi):
+        """Rotation sending a triangle at pole to location theta, phi."""
+        Rth = gs.array([[gs.cos(theta), -gs.sin(theta), 0.],
+                        [gs.sin(theta), gs.cos(theta), 0.],
+                        [0., 0., 1.]])
+        Rphi = gs.array([[gs.cos(phi), 0., gs.sin(phi)],
+                         [0., 1., 0.],
+                         [-gs.sin(phi), 0, gs.cos(phi)]])
+        return Rth @ Rphi @ Rth.transpose()
+
+    def draw_points(self, al=1, zo=0, **kwargs):
+        """Draw points on the Kendall sphere."""
+        points_x = [gs.to_numpy(point[0]) for point in self.points]
+        points_y = [gs.to_numpy(point[1]) for point in self.points]
+        points_z = [gs.to_numpy(point[2]) for point in self.points]
+        self.ax.scatter(points_x, points_y, points_z,
+                        alpha=al, zorder=zo, **kwargs)
+
+    def draw_curve(self, al=1, zo=0, **kwargs):
+        """Draw a curve on the Kendall sphere."""
+        points_x = [gs.to_numpy(point[0]) for point in self.points]
+        points_y = [gs.to_numpy(point[1]) for point in self.points]
+        points_z = [gs.to_numpy(point[2]) for point in self.points]
+        self.ax.plot3D(points_x, points_y, points_z,
+                       alpha=al, zorder=zo, **kwargs)
+
+    def draw_vector(self, v, bp, **kwargs):
```

Comment:
`bp` -> `base_point` (because we use the naming convention `base_point` everywhere else in the code base)

Reply:
Likewise:
`v` -> `tangent_vec`